test/api/teaching_period_test.rb

```diff
@@ -52,4 +72,101 @@ def test_update_break_must_be_from_teaching_period
     assert_equal num_weeks, to_update.number_of_weeks
   end
 
+  # Replace a teaching period
+  def test_put_teaching_period
+    # a dummy teaching period
+    tp = FactoryGirl.create (:teaching_period)
+
+    # data to replace
+    data_to_put = {
+      teaching_period: FactoryGirl.build(:teaching_period),
+      auth_token: auth_token
+    }
+
+    # Update teaching period with id = 1
+    put_json "/api/teaching_periods/#{tp.id}", data_to_put
+    
+    # check if the POST get through
+    assert_equal 200, last_response.status
+
+    # check if the details posted match as expected
+    response_keys = %w(period year start_date end_date active_until)
+    tp_reload = tp.reload
+    assert_json_matches_model(last_response_body, tp_reload, response_keys)
```

Comment:
Hi Andrew, I add test to compare `last_response_body` with `data_to_put` but the details doesnt match as expected. I `puts` the details of two variables and it shows as in the screenshot; it might be that parameters are being compared incorrectly, it seems that 'id' and 'period' are wrongly compared. Can you suggest any possible solution?
![Screen Shot 2020-01-15 at 1 13 50 pm](https://user-images.githubusercontent.com/58777122/72399031-0c0e2900-3799-11ea-880d-eb72a0b8a577.png)